### PR TITLE
feat(anomaly): new anomaly criteria [fixes vm-426]

### PR DESCRIPTION
**Pull request by raftedproc**
## Description
This patch adds new criteria to detect memory limits anomalies to collect statistics. The statistics will be used to adjust the mentioned memory limits. 

## Motivation
AquaVM hardening calls for hard memory limits to be able to fight against DoS. 

## Related Issue(s)
[VM-426](https://linear.app/fluence/issue/VM-426/add-new-criterion-to-anomaly-detection-mechanism-in-nox)

## Proposed Changes
Add new criteria.

## Checklist
- [ ] The code follows the project's coding conventions and style guidelines.
- [x] All tests related to the changes have passed successfully.
- [x] Documentation has been updated to reflect the changes (if applicable).
- [x] All new and existing unit tests have passed.
- [x] I have self-reviewed my code and ensured its quality.
- [x] I have added/updated necessary comments to aid understanding.

## Reviewer Checklist
- [ ] Code has been reviewed for quality and adherence to [guidelines](https://doc.rust-lang.org/1.0.0/style/README.html).
- [ ] Tests have been reviewed and are sufficient to validate the changes.
- [ ] Documentation has been reviewed and is up to date.
- [ ] Any questions or concerns have been addressed.

### CODE REVIEW

File: aquamarine/src/particle_data_store.rs [hunk 1]
```diff
@@ -225,6 +244,21 @@ impl ParticleDataStore {
     }
 }
 
+pub fn get_jvalue_size(value: &JValue) -> usize {
```
Comment (gurinderu) [inline aquamarine/src/particle_data_store.rs:247]: do you need it as pub function ?

File: aquamarine/src/particle_data_store.rs [hunk 2]
```diff
@@ -148,15 +152,30 @@ impl ParticleDataStore {
         Ok(())
     }
 
+    pub fn detect_mem_limits_anomaly(
```
Comment (gurinderu) [inline aquamarine/src/particle_data_store.rs:155]: Also looks like not public fn

File: aquamarine/src/particle_data_store.rs (hunk 1 above)
Comment (raftedproc) [reply aquamarine/src/particle_data_store.rs]: Reduced visibility.

File: aquamarine/src/particle_data_store.rs (hunk 2 above)
Comment (raftedproc) [reply aquamarine/src/particle_data_store.rs]: Reduced visibility.

**Review by gurinderu — APPROVED** [review]

**Pull request merged**